src/bootstrap/builder.rs

```diff
@@ -70,6 +70,8 @@ pub trait Step: 'static + Clone + Debug + PartialEq + Eq + Hash {
     /// Run this rule for all hosts without cross compiling.
     const ONLY_HOSTS: bool = false;
 
+    const PATH: &'static str = "not set";
```

Comment:
I would rather not add this here as it *feels* like the wrong place to put it, could you try instead having it returned as part of `ShouldRun`? I think that should be relatively minimally impactful on the code, just moving a few things around. Calling `should_run` multiple times is fine. I'd also appreciate renaming it to `suite_path` and making it an `Option`.

Please also verify that a mix of specific tests and crates works: `./x.py test src/test/run-pass/foo.rs src/bootstrap` should test both the specific test and bootstrap. I think the current code might not quite do this, but I'm not sure.

Reply:
Thanks! I'm on it